<?xml version="1.0"?>

<launch>
    <arg name="world_name" default="worlds/empty.world"/>
    <arg name="cmd_timeout" default="0.5"/>

    <!-- Vehicle pose -->
    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="z" default="0"/>
    <arg name="roll" default="0.0"/>
    <arg name="pitch" default="0.0"/>
    <arg name="yaw" default="0.0"/>

    <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="mountains.world"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="$(arg use_sim_time)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="headless" value="$(arg headless)"/>
        <arg name="debug" value="$(arg debug)"/>
    </include>

    <include file="$(find clifford_description)/launch/clifford-simple.launch"/>

    <!-- Spawn the vehicle. -->
    <node name="spawn_vehicle" pkg="gazebo_ros" type="spawn_model"
          args="-urdf -param robot_description -model clifford
            -gazebo_namespace /gazebo
            -x $(arg x) -y $(arg y) -z $(arg z)
            -R $(arg roll) -P $(arg pitch) -Y $(arg yaw)"/>

    <!-- Load the joint controllers. One of these publishes the joint states
         to joint_states. -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner"
          args="$(find clifford_gazebo)/config/clifford_simple_joint_ctrlr_params.yaml" output="screen"/>

    <rosparam command="load" file="$(find clifford_gazebo)/config/clifford_mixing_params.yaml"/>
    <node pkg="clifford_gazebo" type="clifford_cmd_mixing.py" name="clifford_cmd_mixing" output="screen"/>


</launch>